adapters/celestia/src/verifier/mod.rs

```diff
@@ -29,12 +33,29 @@ pub const PARITY_SHARES_NAMESPACE: NamespaceId = NamespaceId(hex_literal::hex!("
 impl BlobTransactionTrait for BlobWithSender {
     type Data = BlobIterator;
     type Address = CelestiaAddress;
+
     fn sender(&self) -> CelestiaAddress {
         self.sender.clone()
     }
 
-    fn data(&self) -> Self::Data {
-        self.blob.clone().into_iter()
+    fn data(&self) -> BufWithCounter<Self::Data> {
+        BufWithCounter::new(self.blob.clone().into_iter())
+    }
+
+    fn hash(&self) -> [u8; 32] {
+        // We first try to compute the hash using a hint
+        if let Some(hash) = self.hash {
+            return hash;
+        }
+
+        // If there is no hash computed yet we have to calculate it using the buf data
+        let mut reader = self.data().inner().reader();
+        let mut batch_data = Vec::new();
+        reader
+            .read_to_end(&mut batch_data)
+            .unwrap_or_else(|e| panic!("Unable to read batch data {}", e));
+
+        sha2::Sha256::digest(&batch_data)
```

Comment:
Is it ok to hardcode a specific hash function here?

Reply:
Modified with the last PR: for Celestia set the blob hash as the blob commitment